clocksilencer.ino

```diff
@@ -91,28 +98,25 @@ void ringer(int myhours2) {
   for (int n = 1; n <= myhours2; n++) {
     if (n == 1) {
       myDFPlayer.play(1);
-    } // ring the chime}
+    }  // ring the chime}
     else {
       myDFPlayer.play(2);
     }
     Serial.print("Ring nr:");
     Serial.println(n);
     delay(2250);
   }
-  ~myDFPlayer.begin(); // wondering if it's really necessary here, as we cut the
-                       // power anyway
-  ~hwSerial.begin();
-  digitalWrite(12, 0);
+  digitalWrite(23, 0);
 }
-void deepsleep() {
-  int napTime;
-  if (myMin < 30) {
-    napTime == myMin;
-  } else {
-    napTime = myMin - 30;
-  }
-  esp_sleep_enable_timer_wakeup(napTime * 100000); // in microseconds
+void deepsleep(int sleepHelper) {
+  int countMicro = ((sleepHelper - 1) * 1000000 * 60);
```

Comment:
i thought that you already took care about waking up one minute before e.g. the full hour:

```
sleepHelper = 59 - myMin;
```

so maybe no need to decrement it again by one?

```suggestion
  int countMicro = (sleepHelper * 60 * 1000000);
```

Reply:
Just a precaution 🙂